src/components/publish-site-button.tsx

```diff
@@ -24,30 +28,33 @@ export const PublishSiteButton = () => {
 	} );
 	const { isAnySitePulling, isAnySitePushing } = useSyncSites();
 	const isAnySiteSyncing = isAnySitePulling || isAnySitePushing;
-	const handlePublishClick = () => {
-		setSelectedTab( 'sync' );
+
+	const handlePublishClick = useCallback( () => {
+		if ( redirectToSync ) {
+			setSelectedTab( 'sync' );
+		}
 		dispatch( connectedSitesActions.openModal( 'push' ) );
-	};
+	}, [ redirectToSync, setSelectedTab, dispatch ] );
 
-	if ( connectedSites.length !== 0 ) return null;
+	// Don't show the button if there are already connected sites
+	// (only when redirectToSync is true, meaning it's used outside the sync tab)
+	if ( redirectToSync && connectedSites.length !== 0 ) return null;
 
 	return (
-		<Tooltip
-			disabled={ ! isAnySiteSyncing }
-			text={ __(
-				'Another site is syncing. Please wait for the sync to finish before you publish your site.'
-			) }
-			placement="left"
+		<ConnectButton
+			variant="primary"
+			icon={ cloudUpload }
+			connectSite={ handlePublishClick }
+			disabled={ isAnySiteSyncing }
+			tooltipText={
+				isAnySiteSyncing
+					? __(
+							'Another site is syncing. Please wait for the sync to finish before you publish your site.'
+					  )
+					: __( 'Publishing your site requires an internet connection.' )
+			}
```

Comment:
```suggestion
			disabledTooltipText={ __(
				'Another site is syncing. Please wait for the sync to finish before you publish your site.'
			) }
			offlineTooltipText={ __( 'Publishing your site requires an internet connection.' ) }
```
This is mostly a suggestion, but I think this would clarify how and when which tooltip text is displayed in `ConnectButton`.

Reply:
I like this suggestion but I would delay adding changes to this until https://github.com/Automattic/studio/pull/2171 is merged, at that point we can also refactor the `publish-site-button.tsx` as [suggested here](https://github.com/Automattic/studio/pull/2132#discussion_r2584459482)